core/templates/dev/head/pages/exploration_editor/EditorNavigationDirective.js

```diff
@@ -51,7 +51,7 @@ oppia.directive('editorNavigation', [
               $timeout(function() {
                 $scope.popoverControlObject
                   .postTutorialHelpPopoverIsShown = false;
-              }, 5000);
+              }, 4000);
```

Comment:
Is there any reason why we have this in the first place? Why would first time editor have to wait 4s to start working on exploration?

Reply:
Also I think protractor actually waits for this timeout to be over before proceeding with the next action from the time I use browser.pause() to debug setContent and moveToState

Reply:
This doesn't block the user. It's just a popover in the top right attached to the help menu that says "Click here to get help". I'm fine with protractor waiting for it though.

Reply:
Yup, by the time I said the first post, I realize I made a mistake. The popover was just timing out Protractor's actions

Reply:
Yeah -- I lengthened the timeout to 10 secs. I think it's fine now, but it uncovered a different issue in stateEditor related to the welcome modal not showing up.

The cause of that (which I think I've fixed in the latest commit) is something that's been a bit prevalent now and perhaps could do with a quick look-over. In the e2e tests and test utilities, I keep coming across stuff that uses element.all(), whose return value is plural, but is named in a singular way. This causes problems because doing e.g. waitFor.visibilityOf(x) fails when x is a list of elements found using element.all(). I've fixed these as I came across them, but it might be a good idea to go through the element declarations in the files and make sure that the varnames for element.all() stuff are pluralized.

Reply:
Sounds like a great plan! Let me do a quick audit for it and submit a PR for it.

Reply:
I think this is also the reason why I stopped declaring objects globally like the original [article](https://github.com/angular/protractor/blob/master/docs/page-objects.md) suggested. I kept everything local unless a global usage is needed.

Reply:
Thanks for doing an audit!

Just FYI, I actually think declaring objects globally (within the file) is a good thing. I appreciated that and was wondering why there were a few left out. It should be straightforward to get the names correct, and declaring everything separately makes  the tests easier to read (sort of like separating css from the JS and DOM).